app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsViewModel.kt

```diff
@@ -1,142 +1,119 @@
-                        )
-
-                        database.conversationDao().insert(newConversation)
-                    }
-                    .subscribeOn(Schedulers.io())
-                    .doOnError { t -> Log.w("ConversationViewModel", "Failed to favourite conversation", t) }
-                    .onErrorReturnItem(0)
-                    .subscribe()
-                    .autoDispose()
-        }
+    fun voteInPoll(choices: MutableList<Int>, conversation: ConversationEntity) {
+            viewModelScope.launch {
+                val poll = timelineCases.voteInPoll(conversation.lastStatus.toStatus(), choices).await()
+                val newConversation = conversation.copy(
+                    lastStatus = conversation.lastStatus.copy(poll = poll)
+                )
 
+                database.conversationDao().insert(newConversation)
+            }
     }
 
-    fun expandHiddenStatus(expanded: Boolean, position: Int) {
-        conversations.value?.getOrNull(position)?.let { conversation ->
-            val newConversation = conversation.copy(
+    fun expandHiddenStatus(expanded: Boolean, conversation: ConversationEntity) {
+            viewModelScope.launch {
+                val newConversation = conversation.copy(
                     lastStatus = conversation.lastStatus.copy(expanded = expanded)
-            )
-            saveConversationToDb(newConversation)
-        }
+                )
+                saveConversationToDb(newConversation)
+            }
     }
 
-    fun collapseLongStatus(collapsed: Boolean, position: Int) {
-        conversations.value?.getOrNull(position)?.let { conversation ->
-            val newConversation = conversation.copy(
+    fun collapseLongStatus(collapsed: Boolean, conversation: ConversationEntity) {
+            viewModelScope.launch {
+                val newConversation = conversation.copy(
                     lastStatus = conversation.lastStatus.copy(collapsed = collapsed)
-            )
-            saveConversationToDb(newConversation)
-        }
+                )
+                saveConversationToDb(newConversation)
+            }
     }
 
-    fun showContent(showing: Boolean, position: Int) {
-        conversations.value?.getOrNull(position)?.let { conversation ->
-            val newConversation = conversation.copy(
+    fun showContent(showing: Boolean, conversation: ConversationEntity) {
+            viewModelScope.launch {
+                val newConversation = conversation.copy(
                     lastStatus = conversation.lastStatus.copy(showingHiddenContent = showing)
-            )
-            saveConversationToDb(newConversation)
-        }
+                )
+                saveConversationToDb(newConversation)
+            }
     }
 
-    fun remove(position: Int) {
-        conversations.value?.getOrNull(position)?.let {
-            refresh()
-        }
+    fun remove(conversation: ConversationEntity) {
+        // TODO
```

Comment:
Sill TODO

Reply:
Ah yes. Actually I think one should not be able to delete conversations, because there is no api to actually delete all statuses from a conversation and only deleting the last status is probably not what users expect. Web also doesn't have it, will remove it.

Reply:
Oh wait, there is an Api. 🤔

Reply:
we were always just deleting the newest status in a conversation, so this is actually a nice improvement

Reply:
Ok, this escalated a bit, I fixed deleting conversations and refactored the more menu so it only has the options relevant to conversations (this made a db migration necessary).